### PR TITLE
feat: notarize macOS binaries

### PR DESCRIPTION
**Pull request by bajtos**
- Bump the minimal required macOS version to 10.9 (see [Apple Notarization Requirements](https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution))
- Notarize `zinnia` and `zinniad`

See #23

### CODE REVIEW

File: .github/workflows/ci.yaml
Comment (juliangruber) [inline .github/workflows/ci.yaml:1]: why was this removed?

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -21,9 +23,9 @@ jobs:
       matrix:
         include:
           # List of platforms, this must be in sync with the list of platforms in ci.yaml
-          - target: x86_64-unknown-linux-gnu
-            os: ubuntu-latest
-            name: linux-x64.tar.gz
```
Comment (juliangruber) [inline .github/workflows/release.yml:26]: why was this removed?

**Review by juliangruber — COMMENTED** [review]
is this wip?

**Comment by bajtos** [comment]
Ouch. Enabling the hardened runtime breaks V8. I need to investigate this more.

```
❯ ~/Downloads/archives-macos-x64/zinnia run ~/src/zinnia/mod-peer-checker/peer-checker.js


#
# Fatal error in , line 0
# Check failed: 12 == (*__error()).
#
#
#
#FailureMessage Object: 0x30cf82250
==== C stack trace ===============================



#
# Fatal error in , line 0
# Check failed: 12 == (*__error()).
#
#
#
#FailureMessage Object: 0x30ce7b380
==== C stack trace ===============================

    0   zinnia                              0x0000000102d7b9f3 v8::base::debug::StackTrace::StackTrace() + 19
    1   zinnia                              0x0000000102d8121b v8::platform::(anonymous namespace)::PrintStackTrace() + 27
    2   zinnia                              0x0000000102d714d3 V8_Fatal(char const*, ...) + 323
    3   zinnia                              0x0000000102d7a77d v8::base::OS::SetPermissions(void*, unsigned long, v8::base::OS::MemoryPermission) + 221
    4   zinnia                              0x00000001030075f7 v8::internal::MemoryAllocator::SetPermissionsOnExecutableMemoryChunk(v8::internal::VirtualMemory*, unsigned long, unsigned long, unsigned long) + 199
    5   zinnia                              0x00000001030073e7 v8::internal::MemoryAllocator::AllocateAlignedMemory(unsigned long, unsigned long, unsigned long, v8::internal::AllocationSpace, v8::internal::Executability, void*, v8::internal::VirtualMemory*) + 263
    6   zinnia                              0x000000010300777a v8::internal::MemoryAllocator::AllocateUninitializedChunkAt(v8::internal::BaseSpace*, unsigned long, v8::internal::Executability, unsigned long, v8::internal::PageSize) + 170
    7   zinnia                              0x0000000103007d5f v8::internal::MemoryAllocator::AllocatePage(v8::internal::MemoryAllocator::AllocationMode, v8::internal::Space*, v8::internal::Executability) + 207
    8   zinnia                              0x000000010301b773 v8::internal::PagedSpaceBase::TryExpandBackground(unsigned long) + 51
    9   zinnia                              0x0000000102f41ce0 v8::internal::ConcurrentAllocator::AllocateFromSpaceFreeList(unsigned long, unsigned long, v8::internal::AllocationOrigin) + 192
    10  zinnia                              0x0000000102f41b6f v8::internal::ConcurrentAllocator::AllocateInLabSlow(int, v8::internal::AllocationAlignment, v8::internal::AllocationOrigin) + 47
    11  zinnia                              0x0000000102f6251d v8::internal::LocalHeap::AllocateRaw(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) + 157
    12  zinnia                              0x0000000102f62006 v8::internal::Factory::CodeBuilder::BuildInternal(bool) + 854
    13  zinnia                              0x0000000102e0cf08 v8::internal::baseline::BaselineCompiler::Build(v8::internal::LocalIsolate*) + 360
    14  zinnia                              0x0000000102e0b04f v8::internal::baseline::ConcurrentBaselineCompiler::JobDispatcher::Run(v8::JobDelegate*) + 447
    15  zinnia                              0x0000000102d8089b v8::platform::DefaultJobWorker::Run() + 203
    16  zinnia                              0x0000000102d81a59 v8::platform::DefaultWorkerThreadsTaskRunner::WorkerThread::Run() + 41
    17  zinnia                              0x0000000102d7b2e7 v8::base::ThreadEntry(void*) + 87
    18  libsystem_pthread.dylib             0x00007ff80754b1d3 _pthread_start + 125
    19  libsystem_pthread.dylib             0x00007ff807546bd3 thread_start + 15
[1]    50551 trace trap  ~/Downloads/archives-macos-x64/zinnia run
```

File: .github/workflows/ci.yaml
Comment (bajtos) [reply .github/workflows/ci.yaml:1]: Sorry.

This was not meant to be WIP, but I discovered a critical issue: https://github.com/filecoin-station/zinnia/pull/161#issuecomment-1488745659

I'll revert the first commit after I finish the work.

**Comment by bajtos** [comment]
Progress is blocked by the current GitHub Actions outage 😭

https://www.githubstatus.com/incidents/z3c6q056q332

**Comment by bajtos** [comment]
New workflow run: https://github.com/filecoin-station/zinnia/actions/runs/4555253407

All is good, this PR is ready for final review & landing.

@juliangruber PTAL

```
❯ ~/Downloads/archives-macos-arm64/zinnia run ~/src/zinnia/mod-peer-checker/peer-checker.js
Checking peer /ip4/139.178.91.71/tcp/4001/p2p/QmNnooDu7bfjPFoTZYxMNLWUQJyrVwtbZg5gBMjTezGAJN
Received Identify Message: 1485 bytes
Submitted stats to InfluxDB.
[08:07:17.813 STATS] Jobs completed: 1
^C
```

**Review by juliangruber — APPROVED** [review]
